eng/pipelines/eng-common-sync.yml

```
@@ -30,12 +30,25 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - pwsh: |
+              $patchDir = "$(System.DefaultWorkingDirectory)/patchfiles"
+              $patchfiles = git format-patch HEAD~1...origin/master -- ${{ parameters.DirectoryToSync }} -o $patchDir
```

```suggestion
              $patchfiles = git format-patch HEAD...master -- ${{ parameters.DirectoryToSync }} -o $patchDir
```

Why did you need to back-up one commit?

---

`$(System.PullRequest.SourceBranch)` is actually not present because as you pointed out it is in a detached HEAD state, with HEAD being the pull request merge commit. So backing up one commit is basically going one commit before the merge commit which is what we need.

---

OK you are correct the sourcebranch does not appear in the main repo when the PR is from a fork. So perhaps we should use `refs/remotes/pull/995/merge` or HEAD. I don't think we need to move back one commit.

---

However we should still use the TargetBranch

---

`$(System.PullRequest.TargetBranch)` is just master which does not work as a revision. It needs to be `origin/master`